x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java

```diff
@@ -1230,10 +1254,14 @@ public void testMonitoringUserRole() {
         assertNotNull(roleDescriptor);
         assertThat(roleDescriptor.getMetadata(), hasEntry("_reserved", true));
 
+        final String allowedApplicationActionPattern = "example/custom/action/*";
         Role monitoringUserRole = Role.buildFromRoleDescriptor(
             roleDescriptor,
             new FieldPermissionsCache(Settings.EMPTY),
-            RESTRICTED_INDICES
+            RESTRICTED_INDICES,
+            List.of(
+                new ApplicationPrivilegeDescriptor("kibana-*", "reserved_monitoring", Set.of(allowedApplicationActionPattern), Map.of())
```

Comment:
Good point. I can make this work for sure. `kibana-kibana` doesn't do it because we use randomize below, but using the randomized application name does the trick.